File: Python/Exercicios/Aulas/Pokemon Abstract/pokemonAbstract.py
from abc import ABC, abstractmethod

class PokemonAbstract(ABC):
    def __init__(self, numero: int, nome: str, ataque: int, defesa: int, saude: int, ataqueEspecial: int, defesaEspecial: int, velocidade: int, genero: str) -> None:
        self.numero = numero
        self.nome = nome
        self.ataque = ataque
        self.defesa = defesa
        self.saude = saude
        self.ataqueEspecial = ataqueEspecial
        self.defesaEspecial = defesaEspecial
        self.velocidade = velocidade
        self.genero = genero

    @abstractmethod
    def restaurarVida(self) -> None:
        pass

    @abstractmethod
    def golpear(self) -> None:
        pass

    @abstractmethod
    def defender(self) -> None:
        pass

File: Java/Exercicios/Pokemon/PokemonAbstract.java
public abstract class PokemonAbstract {
    private Integer numero;
    private String nome;
    private Integer ataque;
    private Integer defesa;
    private Integer saude;
    private Integer ataqueEspecial;
    private Integer defesaEspecial;
    private Integer velocidade;
    private String genero;

    public Integer getNumero() {
        return numero;
    }
    public void setNumero(Integer numero) {
        this.numero = numero;
    }

    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }

    public Integer getAtaque() {
        return ataque;
    }
    public void setAtaque(Integer ataque) {
        this.ataque = ataque;
    }

    public Integer getDefesa() {
        return defesa;
    }
    public void setDefesa(Integer defesa) {
        this.defesa = defesa;
    }

    public Integer getSaude() {
        return saude;
    }
    public void setSaude(Integer saude) {
        this.saude = saude;
    }

    public Integer getAtaqueEspecial() {
        return ataqueEspecial;
    }
    public void setAtaqueEspecial(Integer ataqueEspecial) {
        this.ataqueEspecial = ataqueEspecial;
    }

    public Integer getDefesaEspecial() {
        return defesaEspecial;
    }
    public void setDefesaEspecial(Integer defesaEspecial) {
        this.defesaEspecial = defesaEspecial;
    }

    public Integer getVelocidade() {
        return velocidade;
    }
    public void setVelocidade(Integer velocidade) {
        this.velocidade = velocidade;
    }

    public String getGenero() {
        return genero;
    }
    public void setGenero(String genero) {
        this.genero = genero;
    }

    public abstract void restaurarVida();
    public abstract void golpear();
    public abstract void defender();
}